### PR TITLE
HTCONDOR-1757 scitokens cache config

### PR DESCRIPTION
**Pull request by JaimeFrey**
<Insert PR description here, and leave checklist below for code review.>

# HTCondor Pull Request Checklist for internal reviewers

- [ ] Verify that (GitHub thinks) the merge is clean. If it isn't, and you're confident you can resolve the conflicts, do so. Otherwise, send it back to the original developer.
- [ ] Verify that the related Jira ticket exists and has a target version number and that it is correct.
- [ ] Verify that the Jira ticket is in review status and is assigned to the reviewer.
- [ ] Verify that the Jira ticket (HTCONDOR-xxx) is mentioned at the beginning of the title. Edit it, if not
- [ ] Verify that the branch destination of the PR matches the target version of the ticket
- [ ] Check for correctness of change
- [ ] Check for regression test(s) of new features and bugfixes (if the feature doesn't require root)
- [ ] Check for documentation, if needed
- [ ] Check for version history, if needed
- [ ] Check BaTLab dashboard for successful build (https://batlab.chtc.wisc.edu/results/workspace.php) and test for either the PR or a workspace build by the developer that has the Jira ticket as a comment.
- [ ] Check that each commit message references the Jira ticket (HTCONDOR-xxx)

## After the above
- Hit the merge button if the pull request is approved and it is not a security patch (security changes require 2 additional reviews)
- If the pull request is approved, take the ticket out of review state
- Assign JIRA Ticket back to the developer

### CODE REVIEW

**Comment by JaimeFrey** [comment]
This code is based on an unreleased feature of scitokens-cpp. It should not be merged until it can be tested against a new scitokens-cpp release.

File: src/condor_utils/condor_scitokens.cpp [hunk 1]
```diff
@@ -178,6 +183,28 @@ htcondor::init_scitokens()
...
+			}
+			if (!cache_loc.empty()) {
+				cache_loc += "/cache";
+			}
+		}
+		if (!cache_loc.empty()) {
+			dprintf(D_SECURITY|D_VERBOSE, "Setting SciTokens cache directory to %s\n", cache_loc.c_str());
+			char *err = nullptr;
+			if (config_set_str_ptr("keycache.cache_home", cache_loc.c_str(), &err) < 0) {
+				dprintf(D_SECURITY, "Failed to set SciTokens cache directory to %s: %s\n", cache_loc.c_str(), err);
```
[the remark this answers is not in the document]
Comment (JaimeFrey) [reply src/condor_utils/condor_scitokens.cpp]: Probably. I changed it.

**Comment by bbockelm** [comment]
Since we are dlopen’ing the library, shouldn’t the check for the existence of the symbol be done at runtime instead of in the preprocessor?

**Comment by JaimeFrey** [comment]
We don't dlopen the library on macOS or for python wheels. The preprocessor macro check is only for those instances.

**Review by timtheisen — APPROVED** [review]
LGTM

**Pull request merged**